sdk/component/utils/testing/oscf_vulnerability_finding_verification.go

```diff
@@ -0,0 +1,174 @@
+package testing
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/encoding/protojson"
+
+	ocsffindinginfo "github.com/smithy-security/smithy/sdk/gen/ocsf_ext/finding_info/v1"
+	ocsf "github.com/smithy-security/smithy/sdk/gen/ocsf_schema/v1"
+)
+
+// AssertVulnerabilityFindingIsValid uses testify assertions to validate all required fields of a finding.
+// It uses `require` for critical checks that prevent further validation and `assert` for all other checks.
+func AssertVulnerabilityFindingIsValid(t *testing.T, finding *ocsf.VulnerabilityFinding) {
```

Comment:
since we are following the structure of the stretchr APIs, this should return a boolean showing if the check passed or not, so that we can use that in some places to fail immediately and in other places to keep going